### PR TITLE
Template locator

### LINKED ISSUE #12988: Template Engine build is failing after added reference Microsoft.DotNet.TemplateLocator
I switched from the `Microsoft.Internal.DotNet.MockTemplateLocator` to
`Microsoft.DotNet.TemplateLocator` and now the build is failing with the errors:

`Unable to find package Microsoft.DotNet.SdkResolver. No packages exist with this id in source(s): dotnet-eng, dotnet-tools, dotnet5, dotnet5-transport, LocalFeed, nuget.org, nugetbuild`

### PR DESCRIPTION
**Pull request by wli3**
Fix https://github.com/dotnet/sdk/issues/12988

### CODE REVIEW

File: src/Microsoft.DotNet.TemplateLocator/Microsoft.DotNet.TemplateLocator.csproj [hunk 1]
```diff
@@ -12,8 +12,17 @@
...
 
     <PackageId>microsoft.dotnet.templateLocator</PackageId>
+
+    <!--https://github.com/NuGet/Home/issues/3891#issuecomment-377319939-->
+    <TargetsForTfmSpecificBuildOutput>$(TargetsForTfmSpecificBuildOutput);CopyProjectReferencesToPackage</TargetsForTfmSpecificBuildOutput>
   </PropertyGroup>
 
+  <Target Name="CopyProjectReferencesToPackage" DependsOnTargets="ResolveReferences">
+    <ItemGroup>
+      <BuildOutputInPackage Include="@(ReferenceCopyLocalPaths->WithMetadataValue('ReferenceSourceTarget', 'ProjectReference'))" />
```
Comment (joeloff) [inline src/Microsoft.DotNet.TemplateLocator/Microsoft.DotNet.TemplateLocator.csproj:22]: Who/what is consuming this item group? The targetname suggests it's going to copy something, but it seems to just create/add items

File: src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj [hunk 2]
```diff
@@ -8,8 +8,17 @@
...
     <StrongNameKeyId>MicrosoftAspNetCore</StrongNameKeyId>
     <IsPackable>true</IsPackable>
+
+    <!--https://github.com/NuGet/Home/issues/3891#issuecomment-377319939-->
+    <TargetsForTfmSpecificBuildOutput>$(TargetsForTfmSpecificBuildOutput);CopyProjectReferencesToPackage</TargetsForTfmSpecificBuildOutput>
   </PropertyGroup>
 
+  <Target Name="CopyProjectReferencesToPackage" DependsOnTargets="ResolveReferences">
+    <ItemGroup>
+      <BuildOutputInPackage Include="@(ReferenceCopyLocalPaths-&gt;WithMetadataValue('ReferenceSourceTarget', 'ProjectReference'))" />
```
Comment (joeloff) [inline src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj:18]: Any reason we're encoding the "->" operator. Seems like we just use -> in other places?

File: src/Microsoft.DotNet.TemplateLocator/Microsoft.DotNet.TemplateLocator.csproj (hunk 1 above)
Comment (wli3) [reply src/Microsoft.DotNet.TemplateLocator/Microsoft.DotNet.TemplateLocator.csproj:22]: This is the workaround provided by nuget team https://github.com/NuGet/Home/issues/3891#issuecomment-377319939 to include package reference to dll. Nuget targets are consuming this gruop

File: src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj (hunk 2 above)
Comment (wli3) [reply src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj:18]: my VS keep encoding it for some reason. Changing it back

**Review by joeloff — APPROVED** [review]

**Pull request merged**